### PR TITLE
`azurerm_app_configuration_key` - fix test 403 error

### PR DESCRIPTION
**Pull request by teowa**
Introduce a new [`hashicorp/time`](https://registry.terraform.io/providers/hashicorp/time/latest/docs) provider into acctests to sleep 15 min after role assignment, if the new provider is not allowed  here, then the role `App Configuration Data Owner` should be assigned in advance.
from [doc](https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-enable-rbac#azure-built-in-roles-for-azure-app-configuration):
> After a role assignment is made for an identity, allow up to 15 minutes for the permission to propagate before accessing data stored in App Configuration using this identity.

image

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Hi @teowa 

as a rule we should never be sleeping in the provider, this is just a sleep with extra steps and a sign there is something wrong with the resource in question, users should not be waiting after create to use a resource/having to use the time provider like this.

can you please update the create function to poll for the resource to be done propagated?

**Comment by stephybun** [comment]
@teowa any update on this?

File: internal/services/appconfiguration/app_configuration_key_resource.go [hunk 1]
```diff
@@ -138,6 +138,21 @@ func (k KeyResource) Create() sdk.ResourceFunc {
...
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:      []string{"Forbidden"},
+				Target:       []string{"Error", "Exists"},
+				Refresh:      appConfigurationGetKeyRefreshFunc(ctx, client, model.Key, model.Label),
+				PollInterval: 20 * time.Second,
+				Timeout:      15 * time.Minute,
+			}
+
+			if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for App Configuration Key %q read permission to be done propagated: %+v", model.Key, err)
```
Comment (katbyte) [inline internal/services/appconfiguration/app_configuration_key_resource.go:153]:
```suggestion
				return fmt.Errorf("waiting for App Configuration Key %q read permission to be propagated: %+v", model.Key, err)
```

File: internal/services/appconfiguration/app_configuration_feature_resource.go [hunk 2]
```diff
@@ -180,6 +180,22 @@ func (k FeatureResource) Create() sdk.ResourceFunc {
...
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:      []string{"Forbidden"},
+				Target:       []string{"Error", "Exists"},
+				Refresh:      appConfigurationGetKeyRefreshFunc(ctx, client, featureKey, model.Label),
+				PollInterval: 20 * time.Second,
+				Timeout:      15 * time.Minute,
+			}
+
+			if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for App Configuration Key %q read permission to be done propagated: %+v", featureKey, err)
```
Comment (katbyte) [inline internal/services/appconfiguration/app_configuration_feature_resource.go:196]:
```suggestion
				return fmt.Errorf("waiting for App Configuration Key %q read permission to be propagated: %+v", featureKey, err)
```

**Review by katbyte — APPROVED** [review]
Aside from 2 comments LGTM ⛰️

**Pull request merged**